### PR TITLE
Fixed: #20050 Fix Button Alignment of Send Button in Nft-Details

### LINKED ISSUE #20050: Fix Button Alignment of Send Button in Nft-Details
### Describe the bug

In the popup view, the send button is not properly aligned
Screenshot 2023-07-17 at 4 09 42 PM

### Steps to reproduce

1. Go to the NFTs Tab
2. Go to any NFT detail in the popup view

### Error messages or log output

_No response_

### Version

10.33.1

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ducksblock**
## Explanation
The button was not aligned in the NFT details page. I have fixed the issue by assigning it appropriate CSS properties.

* Fixes #20050 

### Before
metamask-before

### After
metamask-after-2

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

**Comment by ducksblock** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ducksblock** [comment]
Hello @NidhiKJha I have changed the CSS properties. Let me know whether this was the expected code change or not.

File: ui/components/app/nft-details/index.scss [hunk 1]
```diff
@@ -112,7 +112,7 @@ $spacer-break-small: 16px;
   }
 
   &__send-button {
-    margin-inline-end: 8px;
+    margin: 8px;
```
Comment (NidhiKJha) [inline ui/components/app/nft-details/index.scss:115]: I think you can remove this margin completely here

File: ui/pages/asset/asset.scss [hunk 2]
```diff
@@ -28,6 +28,6 @@
 
 .asset-options {
   &__button {
-    padding: 2px 0 2px 8px;
+    padding: 2px 4px 2px 4px;
```
[the remark this answers is not in the document]
Comment (NidhiKJha) [reply ui/pages/asset/asset.scss:31]: This change is not required, revert it please

File: ui/components/app/nft-details/nft-details.js [hunk 3]
```diff
@@ -298,8 +297,8 @@ export default function NftDetails({ nft }) {
                     overflowWrap={OverflowWrap.BreakWord}
                     marginBottom={4}
                   >
-                    {`${Number(decWEIToDecETH(lastSale.total_price))} ${
-                      lastSale.payment_token.symbol
+                    {`${Number(decWEIToDecETH(lastSale.total_price))}
+                      ${lastSale.payment_token.symbol
```
Comment (NidhiKJha) [inline ui/components/app/nft-details/nft-details.js:301]: Need to revert this as well, it's causing the lint error in CI for your PR

File: ui/components/app/nft-details/index.scss (hunk 1 above)
Comment (NidhiKJha) [reply ui/components/app/nft-details/index.scss:115]: Updating it to 4px still looks broken
Screenshot 2023-07-25 at 12 59 39 PM

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
I have left some comments here, you should look at the parent of the send button fpr the fix, the margin is getting added there.

**Comment by ducksblock** [comment]
Hey @NidhiKJha first of all I apologize, I am new to open source contributions and working on big repos like this. I am trying my best to understand the code to contribute to Metamask. I have changed the code as per the request and I hope this time it is the correct one, if not then I am down for another try. Thanks!

File: ui/components/app/nft-details/nft-details.js [hunk 4]
```diff
@@ -299,9 +298,8 @@ export default function NftDetails({ nft }) {
                     overflowWrap={OverflowWrap.BreakWord}
                     marginBottom={4}
                   >
-                    {`${Number(decWEIToDecETH(lastSale.total_price))} ${
-                      lastSale.payment_token.symbol
-                    }`}
+                    {`${Number(decWEIToDecETH(lastSale.total_price))} ${lastSale.payment_token.symbol
+                      }`}
```
Comment (NidhiKJha) [inline ui/components/app/nft-details/nft-details.js:302]: Need to fix this lint issue here

File: ui/components/app/nft-details/nft-details.js [hunk 5]
```diff
@@ -459,4 +458,4 @@ NftDetails.propTypes = {
       }),
     }),
   }),
-};
+};
```
Comment (NidhiKJha) [inline ui/components/app/nft-details/nft-details.js:461]: Re: lint fix

File: ui/components/app/nft-details/nft-details.js [hunk 6]
```diff
@@ -425,6 +423,7 @@ export default function NftDetails({ nft }) {
             color={TextColor.textAlternative}
             variant={TextVariant.bodySm}
             as="h6"
+            marginTop={4}
```
Comment (NidhiKJha) [inline ui/components/app/nft-details/nft-details.js:426]: This doesn't fix the alignment here infact it's adding an extra margin top.

**Review by NidhiKJha — COMMENTED** [review]
@ducksblock No worries! I have left some comments.

**Comment by NidhiKJha** [comment]
These changes doesn't seem to fix the issue, you need to look into the Button to add extra margin
Screenshot 2023-07-31 at 2 23 16 PM

File: ui/components/app/nft-details/nft-details.js [hunk 7]
```diff
@@ -145,7 +145,6 @@ export default function NftDetails({ nft }) {
       <Box
         display={Display.Flex}
         width={inPopUp ? BlockSize.Full : BlockSize.Half}
-        margin={inPopUp ? [4, 0] : null}
```
Comment (ducksblock) [inline ui/components/app/nft-details/nft-details.js:148]: @NidhiKJha I removed this and added the marginTop in the other change to make it aligned. I have tested it, it is aligned for me.

Comment (NidhiKJha) [reply ui/components/app/nft-details/nft-details.js:148]: Aah! My bad. Yes, this should fix this. Thanks @ducksblock. Please fix the lint issues to make the CI checks pass and we are good to go.

Comment (ducksblock) [reply ui/components/app/nft-details/nft-details.js:148]: @NidhiKJha It'd be great if you can help me fix these lint errors. As I said I am new to this and these lint errors are so annoying haha

**Review by NidhiKJha — DISMISSED** [review]
The changes look good to me. Thanks for your contribution @ducksblock.

File: ui/components/app/nft-details/nft-details.js (hunk 7 above)
Comment (ducksblock) [reply ui/components/app/nft-details/nft-details.js:148]: @NidhiKJha I have fixed the lint issues, kindly review it. Thanks.

**Comment by NidhiKJha** [comment]
Hey @ducksblock we still have a snapshot failure. Can you please fix this. You can use this command `yarn jest --updateSnapshot`

**Comment by ducksblock** [comment]
> Hey @ducksblock we still have a snapshot failure. Can you please fix this. You can use this command `yarn jest --updateSnapshot`

@NidhiKJha I have fixed the issues.

**Review by NidhiKJha — APPROVED** [review]

**Comment by ducksblock** [comment]
@NidhiKJha Hi, I would like to know if this issue has anything remaining on my side or will it get merged soon?

**Review by vthomas13 — APPROVED** [review]

**Comment by NidhiKJha** [comment]
Hello @ducksblock to merge this PR, all the CI checks need to pass. I think there is a conflict, please refer to this [comment ](https://github.com/MetaMask/metamask-extension/pull/20113#issuecomment-1643910295)by CLA bot. 

Screenshot 2023-08-21 at 2 36 52 PM

**Comment by ducksblock** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ducksblock** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ducksblock** [comment]
> Hello @ducksblock to merge this PR, all the CI checks need to pass. I think there is a conflict, please refer to this [comment ](https://github.com/MetaMask/metamask-extension/pull/20113#issuecomment-1643910295)by CLA bot.
> 
> Screenshot 2023-08-21 at 2 36 52 PM

I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?

**Comment by NidhiKJha** [comment]
> I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?

@ducksblock I think it's best to follow the guidelines to update the email as mentioned in the CLA comment. Hopefully that should fix this issue

**Comment by ducksblock** [comment]
> > I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?
> 
> @ducksblock I think it's best to follow the guidelines to update the email as mentioned in the CLA comment. Hopefully that should fix this issue

I have updated my email as mentioned in the comment.

**Comment by NidhiKJha** [comment]
> > > I think this because I changed my email recently as I no longer use it. Is it possible to fix this issue if I commit again?
> > 
> > 
> > @ducksblock I think it's best to follow the guidelines to update the email as mentioned in the CLA comment. Hopefully that should fix this issue
> 
> I have updated my email as mentioned in the comment.

Since you updated the email can you resign it by commenting `I have read the CLA Document and I hereby sign the CLA`

**Comment by ducksblock** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ducksblock** [comment]
> I have read the CLA Document and I hereby sign the CLA

@NidhiKJha This is not working. Any suggestions are much appreciated.

**Comment by brad-decker** [comment]
Screenshot 2023-09-22 at 11 26 03 AM
Did you happen to commit your last commit on a different machine? it has your name but isn't pulling your profile which might indicate that it thinks that last commit is a new author. You could try to ensure you're logged into your proper machine, your git user name and email is set to the github ones you use, then rebase and force push.

**Pull request closed without merging**